nanoFramework.WebServer/HttpMultipartParser/HeaderUtility.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+
+namespace nanoFramework.WebServer.HttpMultipartParser
+{
+    /// <summary>
+    /// Provides parsing headers from a Http Multipart Form
+    /// </summary>
+    public static class HeaderUtility
+    {
+        /// <summary>
+        /// Reads headers from a line of text.
+        /// Headers are delimited by a semi-colon ';'
+        /// Key-value pairs are separated by colon ':' or equals '='
+        /// Values can be delimited by quotes '"' or not
+        /// </summary>
+        /// <param name="text">The line of text containing one or more headers</param>
+        /// <param name="headers">
+        /// The hashtable that will receive the key values.
+        /// Passed in since a Multipart Part can contain multiple lines of headers
+        /// </param>
+        public static void ParseHeaders(string text, Hashtable headers)
+        {
+            bool inQuotes = false;
+            bool inKey = true;
+            string key = string.Empty;
+            string value = string.Empty;
+
+            foreach (char c in text)
```

Comment:
If you want to parse this:
`Content-Disposition: form-data; name="file"; filename="somefile.json"`

why not something like this, it's simpler and more elegant:
```csharp
var lines = text.Split(':')
 headers.Add(lines[0].ToLower(), string.Empty);
if(lines.Length == 1)
{
  return;
}
var heads = lines[1].Split(';')
foreach(string hd in heads)
{
   var parts = hd.Split('=');
   string key = parts[0].Trim('"');
   string value = string.Empty;
   if (parts.Lenght > 0 )
   { 
       value = string.IsNullOrEmpty(parts[1]) ? string.Empty : parts[1];
    }

   if (!string.IsNullOrEmpty(key))
            {
                headers.Add(key.ToLower(), value);
            }
}
```

Reply:
The implementation resembles the way it was done in the repo I based it on. The main reason to do it this way is that the delimiters can occur in the value(s). For instance:
Content-Disposition: form-data; someExoticKey="bla=bla"; otherKey="bla:bla"; orEven="bla;bla"

Reply:
OK! indeed, your implementation makes sense then!